client/ayon_motionbuilder/api/menu.py

```diff
@@ -4,6 +4,15 @@
 import logging
 from qtpy import QtWidgets, QtCore
 from ayon_core.tools.utils import host_tools
+from ayon_core.settings import get_project_settings
+from ayon_core.pipeline import get_current_project_name
+from .testing import (
+    run_tests_on_repository_workfile,
+    test_create_on_current_workfile,
+    test_publish_on_current_workfile,
+    run_tests
+)
```

Comment:
This has the unfortunate side effect that we now also need to import tests into production code. Negligible perforamnce difference likely but it would be nice if that'd be not needed.

Preferably testing / test suites are more isolated to 'touch' the production code as little as possible.

Reply:
it is possible we can use subprocess to do it or no?

Reply:
If we are going down that route - then this is much better as a `unittest` suite that runs standalone anyway. Which is how tests are usually done anyway. I believe @kalisp also did a lot of work in that area for some other hosts.

What you're doing here now seems much closer to some work @tokejepsen did for maya to just have a "run tests" menu in Maya to, in that way, hopefully lower the barrier for tests for many to run it faster... yet at the same time - it's still up to the dev/user to then start doing that whereas tests preferably end up being possible to automate or to run with a few clicks (which may also mean, not needing to open the host as well).

There's something to say for both.

If, through 'team agreement' the decision is to do what you're doing here - then what you _could_ do is move the import closer to where it's used along the lines of:
```python
if SHOW_TESTS:
    from .tests import a, b, c
```

Anyway, not sure what @antirotor @dee-ynput think? I assume this concept of tests comes from a certain idea to run it manually in hosts?